runtime/bindings/python/hal.cc

```diff
@@ -244,6 +261,37 @@ HalDevice HalDriver::CreateDefaultDevice() {
   return HalDevice::StealFromRawPtr(device);
 }
 
+HalDevice HalDriver::CreateDevice(iree_hal_device_id_t device_id) {
+  // Since the device ids are supposed to be opaque, we need to verify
+  // them by querying available devices.
+  py::list available_devices = QueryAvailableDevices();
```

Comment:
is the worry here that someone in python will change the device id value? like query -> device_id += 1 -> try to create it? is there a way to have a different pybind type that doesn't let users mess with the value instead?

Reply:
Python isn't big on "stopping" people from doing things (even if putting it in a capsule, still pretty mutable). Especially for segfault'y user-visible APIs, some defensiveness is needed.